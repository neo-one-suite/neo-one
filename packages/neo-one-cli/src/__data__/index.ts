export * from './getClients';
export * from './getContracts';
export * from './buildTests';
export * from './generateTests';
