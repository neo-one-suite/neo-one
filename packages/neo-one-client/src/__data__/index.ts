export * from './keys';
export * from './transactions';
