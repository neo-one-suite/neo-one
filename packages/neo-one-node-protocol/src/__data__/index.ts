export { context } from './context';
export { keys } from './keys';
