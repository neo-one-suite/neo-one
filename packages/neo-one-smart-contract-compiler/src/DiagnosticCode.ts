export enum DiagnosticCode {
  UNSUPPORTED_SYNTAX = 0,
  UNKNOWN_TYPE,
  EXPECTED_NUMBER,
  EXPECTED_STRING,
  REFERENCE_ERROR,
  SOMETHING_WENT_WRONG,
  EXPECTED_PROPERTY,
  EXPECTED_CAST,
  TRANSPILATION_ERROR,
  INVALID_CONTRACT_PARAMETER,
  INVALID_CONTRACT_TYPE,
  INVALID_CONTRACT_METHOD,
  INVALID_CONTRACT_EVENT,
  INVALID_VALUE_TYPE,
  INVALID_SYS_CALL,
  UNKNOWN_SYMBOL,
}
