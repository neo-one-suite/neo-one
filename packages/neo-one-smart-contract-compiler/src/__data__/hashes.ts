export const hashes = {
  OLD_NEO_ASSET_HASH: '0xc56f33fc6ecfcd0c225c4ab356fee59390af8560be0e930faebe74a6daff7c9b',
  OLD_GAS_ASSET_HASH: '0x602c79718b16e442de58778e148d0b1084e3b2dffd5de6b7b16cee7969282de7',
};
