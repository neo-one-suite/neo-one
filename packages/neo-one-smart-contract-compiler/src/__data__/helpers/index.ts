export * from './execute';
export * from './extractors';
export * from './startNode';
