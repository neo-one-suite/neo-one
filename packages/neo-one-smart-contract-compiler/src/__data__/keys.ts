import { common, crypto, ECPoint, PrivateKey, PublicKeyString, UInt160 } from '@neo-one/client-common';

export const keys: ReadonlyArray<{
  readonly name: string;
  readonly address: string;
  readonly privateKey: PrivateKey;
  readonly privateKeyString: string;
  readonly publicKey: ECPoint;
  readonly publicKeyString: PublicKeyString;
  readonly wif: string;
  readonly password: string;
  readonly encryptedWIF: string;
  readonly scriptHash: UInt160;
  readonly scriptHashString: string;
}> = [
  {
    name: 'first',
    address: 'NNWAo5vdVJz1oyCuNiaTBA3amBHnWCF4Yk',
    privateKey: common.bufferToPrivateKey(
      Buffer.from('7d128a6d096f0c14c3a25a2b0c41cf79661bfcb4a8cc95aaaea28bde4d732344', 'hex'),
    ),
    privateKeyString: '7d128a6d096f0c14c3a25a2b0c41cf79661bfcb4a8cc95aaaea28bde4d732344',
    publicKey: common.bufferToECPoint(
      Buffer.from('02028a99826edc0c97d18e22b6932373d908d323aa7f92656a77ec26e8861699ef', 'hex'),
    ),
    publicKeyString: common.ecPointToString(
      common.bufferToECPoint(Buffer.from('02028a99826edc0c97d18e22b6932373d908d323aa7f92656a77ec26e8861699ef', 'hex')),
    ),
    wif: 'L1QqQJnpBwbsPGAuutuzPTac8piqvbR1HRjrY5qHup48TBCBFe4g',
    password: 'city of zion',
    encryptedWIF: '6PYLHmDf6AjF4AsVtosmxHuPYeuyJL3SLuw7J1U8i7HxKAnYNsp61HYRfF',
    scriptHash: common.hexToUInt160('36a6e6a7c4658acf096f5d288cc81129c815681c'),
    scriptHashString: '36a6e6a7c4658acf096f5d288cc81129c815681c',
  },
  {
    name: 'second',
    address: 'NiwvMyWYeNghLG8tDyKkWwuZV3wS8CPrrV',
    privateKey: common.bufferToPrivateKey(
      Buffer.from('9ab7e154840daca3a2efadaf0df93cd3a5b51768c632f5433f86909d9b994a69', 'hex'),
    ),
    privateKeyString: '9ab7e154840daca3a2efadaf0df93cd3a5b51768c632f5433f86909d9b994a69',
    publicKey: common.bufferToECPoint(
      Buffer.from('031d8e1630ce640966967bc6d95223d21f44304133003140c3b52004dc981349c9', 'hex'),
    ),
    publicKeyString: common.ecPointToString(
      common.bufferToECPoint(Buffer.from('031d8e1630ce640966967bc6d95223d21f44304133003140c3b52004dc981349c9', 'hex')),
    ),
    wif: 'L2QTooFoDFyRFTxmtiVHt5CfsXfVnexdbENGDkkrrgTTryiLsPMG',
    password: '我的密码',
    encryptedWIF: '6PYWVp3xfgvnuNKP7ZavSViYvvim2zuzx9Q33vuWZr8aURiKeJ6Zm7BfPQ',
    scriptHash: common.hexToUInt160('5461c33e9bbc7de7076754540ba9e62b255ea9fc'),
    scriptHashString: '5461c33e9bbc7de7076754540ba9e62b255ea9fc',
  },
  {
    name: 'third',
    address: 'NTWHAzB82LRGWNuuqjVyyzpGvF3WxbbPoG',
    privateKey: common.bufferToPrivateKey(
      Buffer.from('3edee7036b8fd9cef91de47386b191dd76db2888a553e7736bb02808932a915b', 'hex'),
    ),
    privateKeyString: '3edee7036b8fd9cef91de47386b191dd76db2888a553e7736bb02808932a915b',
    publicKey: common.bufferToECPoint(
      Buffer.from('02232ce8d2e2063dce0451131851d47421bfc4fc1da4db116fca5302c0756462fa', 'hex'),
    ),
    publicKeyString: common.ecPointToString(
      common.bufferToECPoint(Buffer.from('02232ce8d2e2063dce0451131851d47421bfc4fc1da4db116fca5302c0756462fa', 'hex')),
    ),
    wif: 'KyKvWLZsNwBJx5j9nurHYRwhYfdQUu9tTEDsLCUHDbYBL8cHxMiG',
    password: 'MyL33tP@33w0rd',
    encryptedWIF: '6PYNoc1EG5J38MTqGN9Anphfdd6UwbS4cpFCzHhrkSKBBbV1qkbJJZQnkn',
    scriptHash: common.hexToUInt160('a38db239bb36ace1a69eb727a5b0157baa094653'),
    scriptHashString: 'a38db239bb36ace1a69eb727a5b0157baa094653',
  },
];

export const addKeysToCrypto = () => {
  keys.forEach(({ privateKey, publicKey }) => {
    crypto.addPublicKey(privateKey, publicKey);
  });
};
