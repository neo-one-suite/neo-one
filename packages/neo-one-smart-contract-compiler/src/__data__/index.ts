import * as helpers from './helpers';

export { helpers };
export * from './keys';
