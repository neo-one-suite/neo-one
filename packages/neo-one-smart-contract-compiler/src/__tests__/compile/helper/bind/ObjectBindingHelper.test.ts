import { helpers } from '../../../../__data__';
import { DiagnosticCode } from '../../../../DiagnosticCode';

describe('ObjectBindingHelper', () => {
  describe('variable', () => {
    test('simple', async () => {
      await helpers.executeString(`
        const value = { a: 0, b: 1, c: 2, 3: 3, 4: [4] };
        const x = 4;
        const { a, b, 3: three, [x]: [four], ...rest } = value;

        assertEqual(a, 0);
        assertEqual(b, 1);
        assertEqual(Object.keys(rest).length, 1);
        assertEqual(rest.c, 2);
        assertEqual(three, 3);
        assertEqual(four, 4);
      `);
    });

    test('builtin value', async () => {
      await helpers.executeString(`
        import { Blockchain } from '@neo-one/smart-contract';
        const { currentTransaction = Blockchain.currentTransaction } = Blockchain;

        assertEqual(currentTransaction !== undefined, true);
      `);
    });

    test('builtin value string property', async () => {
      await helpers.executeString(`
        import { Blockchain } from '@neo-one/smart-contract';
        const { 'currentTransaction': currentTransaction } = Blockchain;

        assertEqual(currentTransaction !== undefined, true);
      `);
    });

    test('builtin value computed property', async () => {
      await helpers.compileString(
        `
        import { Blockchain } from '@neo-one/smart-contract';
        const { ['currentTransaction']: currentTransaction } = Blockchain;

        assertEqual(currentTransaction !== undefined, true);
      `,
        { type: 'error', code: DiagnosticCode.UnsupportedSyntax },
      );
    });

    test('builtin value rest property', async () => {
      await helpers.compileString(
        `
        import { Blockchain } from '@neo-one/smart-contract';
        const { currentTransaction, ...rest } = Blockchain;

        assertEqual(currentTransaction !== undefined, true);
      `,
        { type: 'error', code: DiagnosticCode.UnsupportedSyntax },
      );
    });

    test('builtin value unknown property', async () => {
      await helpers.compileString(
        `
        import { Blockchain } from '@neo-one/smart-contract';
        const { 0: something } = Blockchain;

        assertEqual(something !== undefined, true);
      `,
        { type: 'error' },
      );
    });

    test('builtin value method property', async () => {
      await helpers.compileString(
        `
        import { Address } from '@neo-one/smart-contract';
        const { isSender } = Address;

        assertEqual(isSender !== undefined, true);
      `,
        { type: 'error', code: DiagnosticCode.InvalidBuiltinReference },
      );
    });

    test('builtin value instance computed property', async () => {
      await helpers.compileString(
        `
        import { Blockchain } from '@neo-one/smart-contract';
        const { ['signers']: signers } = Blockchain.currentTransaction;

        assertEqual(signers !== undefined, true);
      `,
        { type: 'error', code: DiagnosticCode.UnsupportedSyntax },
      );
    });

    test('builtin instance value method property', async () => {
      await helpers.compileString(
        `
        import { ArrayStorage } from '@neo-one/smart-contract';
        const { forEach } = ArrayStorage.for<number>();

        assertEqual(forEach !== undefined, true);
      `,
        { type: 'error', code: DiagnosticCode.InvalidBuiltinReference },
      );
    });
  });

  describe('function parameter', () => {
    test('simple', async () => {
      await helpers.executeString(`
        const value = { a: 0, b: 1, c: 2 };

        function run({ a, b, ...rest }: typeof value) {
          assertEqual(a, 0);
          assertEqual(b, 1);
          assertEqual(Object.keys(rest).length, 1);
          assertEqual(rest.c, 2);
        }

        run(value);
      `);
    });
  });

  describe('arrow function parameter', () => {
    test('simple', async () => {
      await helpers.executeString(`
        const value = { a: 0, b: 1, c: 2 };

        const run = ({ a, b, ...rest }: typeof value) => {
          assertEqual(a, 0);
          assertEqual(b, 1);
          assertEqual(Object.keys(rest).length, 1);
          assertEqual(rest.c, 2);
        }

        run(value);
      `);
    });
  });

  describe('constructor parameter', () => {
    test('simple', async () => {
      await helpers.executeString(`
        const value = { a: 0, b: 1, c: 2 };

        class Foo {
          constructor({ a, b, ...rest }: typeof value) {
            assertEqual(a, 0);
            assertEqual(b, 1);
            assertEqual(Object.keys(rest).length, 1);
            assertEqual(rest.c, 2);
          }
        }

        new Foo(value);
      `);
    });
  });

  describe('method parameter', () => {
    test('simple', async () => {
      await helpers.executeString(`
        const value = { a: 0, b: 1, c: 2 };

        class Foo {
          public foo({ a, b, ...rest }: typeof value) {
            assertEqual(a, 0);
            assertEqual(b, 1);
            assertEqual(Object.keys(rest).length, 1);
            assertEqual(rest.c, 2);
          }
        }

        new Foo().foo(value);
      `);
    });
  });
});
