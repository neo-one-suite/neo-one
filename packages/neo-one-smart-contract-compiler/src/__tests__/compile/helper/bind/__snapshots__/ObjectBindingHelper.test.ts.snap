// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObjectBindingHelper variable builtin instance value method property 1`] = `
"snippetCode.ts (3,17): Builtin properties cannot be referenced

      1 | 
      2 |         import { ArrayStorage } from '@neo-one/smart-contract';
    > 3 |         const { forEach } = ArrayStorage.for<number>();
        |                 ^
      4 | 
      5 |         assertEqual(forEach !== undefined, true);
      6 |       
"
`;

exports[`ObjectBindingHelper variable builtin value computed property 1`] = `
"snippetCode.ts (3,17): Unsupported syntax.

      1 | 
      2 |         import { Blockchain } from '@neo-one/smart-contract';
    > 3 |         const { ['currentTransaction']: currentTransaction } = Blockchain;
        |                 ^
      4 | 
      5 |         assertEqual(currentTransaction !== undefined, true);
      6 |       
"
`;

exports[`ObjectBindingHelper variable builtin value instance computed property 1`] = `
"snippetCode.ts (3,17): Unsupported syntax.

      1 | 
      2 |         import { Blockchain } from '@neo-one/smart-contract';
    > 3 |         const { ['signers']: signers } = Blockchain.currentTransaction;
        |                 ^
      4 | 
      5 |         assertEqual(signers !== undefined, true);
      6 |       
"
`;

exports[`ObjectBindingHelper variable builtin value method property 1`] = `
"snippetCode.ts (3,17): Builtin properties cannot be referenced

      1 | 
      2 |         import { Address } from '@neo-one/smart-contract';
    > 3 |         const { isSender } = Address;
        |                 ^
      4 | 
      5 |         assertEqual(isSender !== undefined, true);
      6 |       
"
`;

exports[`ObjectBindingHelper variable builtin value rest property 1`] = `
"snippetCode.ts (3,37): Unsupported syntax. This is not supported because it would result in inefficient smart contract code

      1 | 
      2 |         import { Blockchain } from '@neo-one/smart-contract';
    > 3 |         const { currentTransaction, ...rest } = Blockchain;
        |                                     ^
      4 | 
      5 |         assertEqual(currentTransaction !== undefined, true);
      6 |       
"
`;

exports[`ObjectBindingHelper variable builtin value unknown property 1`] = `
"snippetCode.ts (3,17): Property '0' does not exist on type 'BlockchainConstructor'.

      1 | 
      2 |         import { Blockchain } from '@neo-one/smart-contract';
    > 3 |         const { 0: something } = Blockchain;
        |                 ^
      4 | 
      5 |         assertEqual(something !== undefined, true);
      6 |       
"
`;
